### PR TITLE
feat: Packaging with Hatch

### PR DESCRIPTION
**Pull request by nimbol**

### CODE REVIEW

File: .github/workflows/ci-test.yaml [hunk 1]
```diff
@@ -68,13 +70,13 @@ jobs:
...
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v3
 
     - name: Install dependencies
-      run: |
-        python -m venv venv
-        source venv/bin/activate
-        pip install -r requirements-all.txt
+      run: uv sync --all-extras
```
Comment (dswistowski) [inline .github/workflows/ci-test.yaml:79]: you do not want to update lock file
```suggestion
      run: uv sync --all-extras --frozen
```

File: .github/workflows/ci-test.yaml [hunk 2]
```diff
@@ -68,13 +70,13 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@v5
       with:
-        python-version: '3.x'
+        python-version: '3.8'
```
Comment (dswistowski) [inline .github/workflows/ci-test.yaml:73]: maybe it's worth to do matrix test against python 3.8, 3.10, 3.12 ?

File: .github/workflows/ci-test.yaml [hunk 3]
```diff
@@ -14,6 +14,8 @@ jobs:
     steps:
     - name: Checkout code
       uses: actions/checkout@v4
+      with:
+        fetch-tags: true
```
Comment (dswistowski) [inline .github/workflows/ci-test.yaml:18]: you might need `fetch-depth: 0`

File: .github/workflows/ci-test.yaml (hunk 2 above)
Comment (nimbol) [reply .github/workflows/ci-test.yaml:74]: Seems like a good idea. Let's check that publish works first. I've tumbled down a bit of a rabbit hole in this repo today and would prefer to make no further changes than necessary.

File: pyproject.toml [hunk 4]
```diff
@@ -1,12 +1,64 @@
...
+# Example formatted version: 1.2.3
+
+distance-dirty = "{base_version}.post{distance}"
+# Example formatted version: 1.2.3.post42
 
 [tool.ruff]
-src = ["rated_exporter_sdk", "tests"]
+src = ["src", "tests"]
 ignore = ["PLR2004", "E722", "PLR0913", "G004", "C901", "PLR0912", "PLR0915"]
 line-length = 125
```
[the remark this answers is not in the document]
Comment (nimbol) [reply pyproject.toml:59]: It seems that `black` has the last say in this. I'm not sure how much effect this setting for `ruff` is having right now.

[the remark this answers is not in the document]
Comment (dswistowski) [reply pyproject.toml:59]: Black is 88 if you do not change

**Review by dswistowski — APPROVED** [review]

**Pull request merged**